### PR TITLE
[17.0] [MIG] sale_order_line_date: migration to 17.0

### PR DESCRIPTION
**Pull request by siemenv**

### CODE REVIEW

File: sale_order_line_date/models/sale_order_line.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            )
+        return vals
+
+    def write(self, vals):
+        res = super().write(vals)
+        moves_to_upd = set()
+        if "commitment_date" in vals:
+            for move in self.move_ids:
+                if move.state not in ["cancel", "done"]:
+                    moves_to_upd.add(move.id)
```
Comment (celm1990) [inline sale_order_line_date/models/sale_order_line.py]: `write` work over multi records, please avoid singleton error if write is done with multi records
```suggestion
            for line in self:
              for move in line.move_ids:
                  if move.state not in ["cancel", "done"]:
                      moves_to_upd.add(move.id)
```

**Review by celm1990 — CHANGES_REQUESTED** [review]

File: sale_order_line_date/models/sale_order_line.py (hunk 1 above)
Comment (yannoliv) [reply sale_order_line_date/models/sale_order_line.py]: Both self and move_ids are recordsets, so I think this code is viable as self.move_ids will just get all move_ids of all records in self. @celm1990

**Comment by bizzappdev** [comment]
Thank you for the contribution and fix.
Functional test okay.
Code LGTM.
Only the commit message should follow the OCA standard 
$ git commit -m "[MIG] $module: Migration to 17.0"
Please make the necessary changes.

**Comment by salvorapi** [comment]
LGTM

**Review by celm1990 — APPROVED** [review]
LGTM

File: sale_order_line_date/models/sale_order_line.py (hunk 1 above)
Comment (BialArno) [reply sale_order_line_date/models/sale_order_line.py]: Was wondering, why iterate when you can map/filter instead?

```python
        if 'commitment_date' in values and (moves_to_update := self.mapped('move_ids').filtered(lambda move: move.state not in ['cancel', 'done'])):
            moves_to_update.date_deadline = values.get('commitment_date')
```

Comment (rousseldenis) [reply sale_order_line_date/models/sale_order_line.py]: @BialArno In fact, mapped.filtered do the same as the current code. I was using Odoo provided methods before but now I'd prefer this writing as clearer.

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by rousseldenis** [comment]
@celm1990 @yannoliv Do you want to update your review ?

**Review by bizzappdev — APPROVED** [review]
LGTM Code review and functional testing both

File: sale_order_line_date/models/sale_order_line.py (hunk 1 above)
Comment (BialArno) [reply sale_order_line_date/models/sale_order_line.py]: @rousseldenis well yes, they do the same, more or less it's pick of personal preferrence, but then we should consider implementing framework methods over sticking to python basics or go plain python and at that point, in my opinion, in Odoo these methods weren't added for nothing :) Ofc competition of speed might be useful to check if for loop or mapped/filtered does same task faster and/or more resource efficient.

**Review by celm1990 — APPROVED** [review]
LGTM

**Review by DorianMAG — APPROVED** [review]
the code looks good
Can't test in runboat to make a functional review

**Review by salvorapi — APPROVED** [review]

**Review by Deriman-Alonso — APPROVED** [review]
LGTM 👍, code review, Thanks!

**Review by traitrantrobz — APPROVED** [review]
The test file and functional is LGTM 👍
Please rebase the commit.
Ảnh chụp Màn hình 2024-07-22 lúc 11 40 18

**Comment by HaraldPanten** [comment]
@siemenv Could you "clean" the commit history a little bit?

THX

**Review by PieterPaulussen — APPROVED** [review]

**Review by BT-etejeda — APPROVED** [review]
Code and functional review!
@siemenv could you please squash the administrative commits?
If you don't have time to do it, let us now, please. Thanks.

**Comment by StefanRijnhart** [comment]
I do apologize but with 9 approvals I think we can live with a handful of redundant administrative commits.

/ocabot merge nobump

**Pull request merged**